### PR TITLE
In 783 add pancan assay

### PR DESCRIPTION
**Pull request by PolarisAstrum-NHS**
Added pancan assay and visualisation of RNAseq metrics

- - -
This change is [Reviewable](https://reviewable.io/reviews/eastgenomics/trendyQC/162)

## Summary by CodeRabbit

- **New Features**
  - Introduced a new section "PanCancer" in the configuration for MultiQC metrics.
  - Added new metrics related to RNA sequencing in various configuration files.
  - New classes for RNA sequencing metrics and quality control have been created.
  - New configuration file for RNA sequencing quality control metrics added.

- **Improvements**
  - Enhanced error handling and feedback in command outputs and report generation.
  - Centralised colour management for plotting through a new settings dictionary.

- **Bug Fixes**
  - Improved clarity of error messages and traceback outputs for easier debugging.

### CODE REVIEW

**Comment by mattgarner** [comment]
@coderabbitai review

File: trendyqc/trend_monitoring/backend_utils/plot.py [hunk 1]
```diff
@@ -300,32 +301,7 @@ def format_data_for_plotly_js(plot_data: pd.DataFrame) -> tuple:
...
-            "#00FF00",  # bright green
-        ],
-        "Twist WES": [
-            "#ff65ff",  # fushia
-            "#6600cc",  # purple
-            "#1c6dff",  # blue
-            "#6ddfff",  # light blue
-        ],
-    }
+    colors = settings.PLOTTING_COLORS
```
[the remark this answers is not in the document]
Comment (PolarisAstrum-NHS) [reply trendyqc/trend_monitoring/backend_utils/plot.py]: Done.

File: trendyqc/trend_monitoring/backend_utils/plot.py [hunk 2]
```diff
@@ -300,32 +302,17 @@ def format_data_for_plotly_js(plot_data: pd.DataFrame) -> tuple:
...
+        colors = settings.PLOTTING_COLORS
+    except AttributeError:
+        raise ImproperlyConfigured(
+            "PLOTTING_COLORS setting is missing. Please define it in your settings file."
+        )
+
+    if not isinstance(colors, dict):
+        raise ImproperlyConfigured(
+            "PLOTTING_COLORS must be a dictionary mapping assay names to colour lists."
+        )
```
[the remark this answers is not in the document]
Comment (PolarisAstrum-NHS) [reply trendyqc/trend_monitoring/backend_utils/plot.py:315]: Done.

File: trendyqc/trend_monitoring/management/commands/add_projects.py [hunk 3]
```diff
@@ -198,3 +198,5 @@ def handle(self, *args, **options):
                 logger.warning(warning_report)
 
             slack_notify(summary_report)
+
+        print(final_msg)
```
[the remark this answers is not in the document]
Comment (PolarisAstrum-NHS) [reply trendyqc/trend_monitoring/management/commands/add_projects.py]: Done.

File: trendyqc/trend_monitoring/management/commands/utils/_multiqc.py [hunk 4]
```diff
@@ -596,9 +596,10 @@ def import_instances(self):
                     instance_model_name = type(instance).__name__
                     msg = (
                         "Could not be imported because of "
-                        f"{instance_model_name}:\n{e}"
+                        f"`{instance_model_name}`:\n```{e}```"
                     )
                     self.messages.append((msg, "error"))
+                    print(traceback.format_exc())
```
[the remark this answers is not in the document]
Comment (PolarisAstrum-NHS) [reply trendyqc/trend_monitoring/management/commands/utils/_multiqc.py]: Done.

[the remark this answers is not in the document]
Comment (PolarisAstrum-NHS) [reply trendyqc/trend_monitoring/management/commands/utils/_multiqc.py]: Done.

File: trendyqc/trend_monitoring/management/commands/utils/_multiqc.py [hunk 5]
```diff
@@ -596,9 +597,11 @@ def import_instances(self):
                     instance_model_name = type(instance).__name__
                     msg = (
                         "Could not be imported because of "
-                        f"{instance_model_name}:\n{e}"
+                        f"`{instance_model_name}`:\n```{e}```"
                     )
                     self.messages.append((msg, "error"))
+                    logger = logging.getLogger(__name__)
+                    logger.error("Failed to import instance: %s", traceback.format_exc())
```
[the remark this answers is not in the document]
Comment (PolarisAstrum-NHS) [reply trendyqc/trend_monitoring/management/commands/utils/_multiqc.py:604]: Done.

File: trendyqc/trend_monitoring/management/commands/utils/_report.py [hunk 6]
```diff
@@ -60,6 +60,7 @@ def setup_report_object(project_id: list):
                     multiqc_job_id=job_id,
                 )
                 multiqc_report.add_msg(msg)
+                print(traceback.format_exc())
```
[the remark this answers is not in the document]
Comment (PolarisAstrum-NHS) [reply trendyqc/trend_monitoring/management/commands/utils/_report.py]: Done.

File: trendyqc/trend_monitoring/management/commands/utils/_report.py [hunk 7]
```diff
@@ -78,7 +79,7 @@ def import_multiqc_report(report: MultiQC_report):
         try:
             report.import_instances()
         except Exception:
-            msg = f"Failed to import\n" "```" f"{traceback.format_exc()}" "```"
+            msg = f"Failed to import\n```{traceback.format_exc()}```"
```
[the remark this answers is not in the document]
Comment (PolarisAstrum-NHS) [reply trendyqc/trend_monitoring/management/commands/utils/_report.py]: Done.

File: trendyqc/trend_monitoring/management/configs/assays.json [hunk 8]
```diff
@@ -82,5 +82,11 @@
         "multiqc_sentieon_AlignmentSummaryMetrics": ["picard", "alignment_summary_metrics"],
         "multiqc_sentieon_insertSize": ["picard", "insert_size_metrics"],
         "multiqc_verifybamid": ["verifybamid", null]
+    },
+    "PanCancer": {
+        "multiqc_fastqc": ["fastqc", "read_data"],
+        "multiqc_general_stats": ["custom_coverage", null],
+        "multiqc_picard_RnaSeqMetrics": ["picard", "RNAseq_metrics"],
+        "multiqc_rna_seqc": ["rna_seqc", null]
```
[the remark this answers is not in the document]
Comment (PolarisAstrum-NHS) [reply trendyqc/trend_monitoring/management/configs/assays.json:90]: Done.

File: trendyqc/trend_monitoring/management/configs/tool_configs/picard.json [hunk 9]
```diff
@@ -213,5 +213,35 @@
         "READS_IN_DUPLICATE_PAIRS_NONOPTICAL": "reads_in_duplicate_pairs_nonoptical",
         "READS_IN_DUPLICATE_UNPAIRED": "reads_in_duplicate_unpaired",
         "READS_UNMAPPED": "reads_unmapped"
+    },
+    "RNAseq_metrics": {
+        "PF_BASES": "pf_bases",
```
[the remark this answers is not in the document]
Comment (PolarisAstrum-NHS) [reply trendyqc/trend_monitoring/management/configs/tool_configs/picard.json:218]: Done.

File: trendyqc/trend_monitoring/management/configs/tool_configs/rna_seqc.json [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+    "Ambiguous Alignment Rate": "ambiguous_alignment_rate",
+    "Ambiguous Reads": "ambiguous_reads",
+    "Average Fragment Length": "average_fragment_length",
+    "Avg. Splits per Read": "avg_splits_per_read",
+    "Base Mismatch": "base_mismatch",
+    "Chimeric Alignment Rate": "chimeric_alignment_rate",
+    "Chimeric Reads": "chimeric_reads",
+    "Discard Rate": "discard_rate",
+    "Duplicate Rate of Mapped": "duplicate_rate_of_mapped",
+    "Duplicate Rate of Mapped, excluding Globins": "duplicate_rate_of_mapped_exclusing_globins",
```
[the remark this answers is not in the document]
Comment (PolarisAstrum-NHS) [reply trendyqc/trend_monitoring/management/configs/tool_configs/rna_seqc.json]: Done.

File: trendyqc/trend_monitoring/management/configs/tool_configs/rna_seqc.json [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+    "Sample": "sample",
+    "Total Bases": "total_bases",
+    "Total Mapped Pairs": "total_mapped_pairs",
+    "Total Read Number": "total_read_number",
+    "Unique Mapping, Vendor QC Passed Reads": "unique_mapping_vendor_qc_passed_reads",
+    "Unique Rate of Mapped": "unique_rate_of_mapped",
+    "Unpaired Reads": "unpaired_reads",
+    "rRNA Reads": "rrna_reads",
+    "rRNA rate": "rrna_rate"
+}
```
[the remark this answers is not in the document]
Comment (PolarisAstrum-NHS) [reply trendyqc/trend_monitoring/management/configs/tool_configs/rna_seqc.json]: Done.

File: trendyqc/trend_monitoring/models/bam_qc.py [hunk 12]
```diff
@@ -145,6 +145,9 @@ class Picard(models.Model):
         blank=True,
         null=True,
     )
+    rnaseq_metrics = models.ForeignKey(
+        "RNAseq_metrics", on_delete=models.DO_NOTHING, blank=True, null=True
```
[the remark this answers is not in the document]
Comment (PolarisAstrum-NHS) [reply trendyqc/trend_monitoring/models/bam_qc.py]: Done.

File: trendyqc/trend_monitoring/models/bam_qc.py [hunk 13]
```diff
@@ -406,3 +409,127 @@ class Duplication_metrics(models.Model):
...
+    pct_correct_strand_reads = models.FloatField()
+    median_cv_coverage = models.FloatField()
+    median_5prime_bias = models.FloatField()
+    median_3prime_bias = models.FloatField()
+    median_5prime_to_3prime_bias = models.FloatField()
+    pf_not_aligned_bases = models.FloatField()
+
+    class Meta:
+        app_label = "trend_monitoring"
+        db_table = "rnaseq_metrics"
```
[the remark this answers is not in the document]
Comment (PolarisAstrum-NHS) [reply trendyqc/trend_monitoring/models/bam_qc.py:446]: Done.

File: trendyqc/trend_monitoring/models/bam_qc.py [hunk 14]
```diff
@@ -406,3 +409,127 @@ class Duplication_metrics(models.Model):
...
+    average_fragment_length = models.FloatField()
+    fragment_length_median = models.FloatField()
+    fragment_length_std = models.FloatField()
+    fragment_length_mad_std = models.FloatField()
+    median_of_avg_transcript_coverage = models.FloatField()
+    median_of_transcript_coverage_std = models.FloatField()
+    median_of_transcript_coverage_cv = models.FloatField()
+    median_exon_cv = models.FloatField()
+    exon_cv_mad = models.FloatField()
+    sample = models.CharField()
```
[the remark this answers is not in the document]
Comment (PolarisAstrum-NHS) [reply trendyqc/trend_monitoring/models/bam_qc.py:531]: Done.

File: trendyqc/trend_monitoring/models/bam_qc.py [hunk 15]
```diff
@@ -406,3 +409,127 @@ class Duplication_metrics(models.Model):
...
+
+    class Meta:
+        app_label = "trend_monitoring"
+        db_table = "rna_seqc"
```
[the remark this answers is not in the document]
Comment (PolarisAstrum-NHS) [reply trendyqc/trend_monitoring/models/bam_qc.py]: Done.

File: trendyqc/trend_monitoring/models/metadata.py [hunk 16]
```diff
@@ -59,6 +59,9 @@ class Report_Sample(models.Model):
     picard = models.ForeignKey(
         "Picard", on_delete=models.DO_NOTHING, blank=True, null=True
     )
+    rna_seqc = models.ForeignKey(
+        "RNA_seqc", on_delete=models.DO_NOTHING, blank=True, null=True
+    )
```
[the remark this answers is not in the document]
Comment (PolarisAstrum-NHS) [reply trendyqc/trend_monitoring/models/metadata.py:64]: Done.

File: trendyqc/trendyqc/settings.py [hunk 17]
```diff
@@ -311,3 +311,37 @@
...
+        "#1c6dff",  # blue
+        "#6ddfff",  # light blue
+    ],
+    "PanCancer": [
+        "#a7cecb",  # light blue
+        "#f2f230",  # yellow
+        "#8447ff",  # purple
+        "#2b4141",  # dark green
+    ],
+}
```
[the remark this answers is not in the document]
Comment (PolarisAstrum-NHS) [reply trendyqc/trendyqc/settings.py:347]: Done.

**Review by Yu-jinKim — APPROVED** [review]
> Reviewed 12 of 12 files at r1, 5 of 6 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/trendyQC/162)* status: all files reviewed, 18 unresolved discussions (waiting on @PolarisAstrum-NHS)

**Pull request merged**